src/coreclr/nativeaot/Runtime/RuntimeInstance.cpp

```diff
@@ -66,18 +66,22 @@ PTR_UInt8 RuntimeInstance::FindMethodStartAddress(PTR_VOID ControlPC)
     return NULL;
 }
 
-ICodeManager * RuntimeInstance::FindCodeManagerByAddress(PTR_VOID pvAddress)
+// WARNING: This method is called by suspension while one thread is interrupted
+//          in a random location, possibly holding random locks.
+//          It is unsafe to use blocking APIs or allocate in this method.
+//          Please ensure that all methods called by this one also have this warning.
+bool RuntimeInstance::IsManaged(PTR_VOID pvAddress)
 {
-    ReaderWriterLock::ReadHolder read(&m_ModuleListLock);
+    return m_CodeManager->IsManaged(pvAddress);
```

Comment:
I think ability to check if an IP is in managed code is important. The check should be fast and simple (since we can't take locks or allocate). The original implementation was walking a linked list while taking a read lock.

If we would want to support multiple code managers, I'd prefer a model when adding another code manager merges its data into some lookup data structure that is optimized for lookup.

I think that would be like moving `IsManaged` API to the `RuntimeInstance` and for now implementing it by storing the range in the instance once a code manager is added..

Reply:
> If we would want to support multiple code managers, I'd prefer a model when adding another code manager merges its data into some lookup data structure that is optimized for lookup.

Yes, that's fine. The knowledge of the address ranges managed by each code manager is still in the runtime instance in this model.